File: lib/app/Models/GetPriceWithVatDTO.php
<?php
/**
 * GetPriceWithVatDTO
 */
namespace app\Models;

/**
 * GetPriceWithVatDTO
 * @description Цена с указанием ставки НДС и времени последнего обновления.
 */
class GetPriceWithVatDTO {

    /** @var float $value Цена на товар.*/
    public $value = 0;

    /** @var float $discountBase Цена на товар без скидки.  Число должно быть целым. Вы можете указать цену со скидкой от 5 до 99%.  Передавайте этот параметр при каждом обновлении цены, если предоставляете скидку на товар.*/
    public $discountBase = 0;

    /** @var string $currencyId */
    public $currencyId = "";

    /** @var int $vat Идентификатор ставки НДС, применяемой для товара:  * &#x60;2&#x60; — 10%. * &#x60;5&#x60; — 0%. * &#x60;6&#x60; — не облагается НДС. * &#x60;7&#x60; — 20%.  Если параметр не указан, используется ставка НДС, установленная в кабинете.*/
    public $vat = 0;

    /** @var \DateTime $updatedAt Время последнего обновления.*/
    public $updatedAt;

}

File: lib/Models/GetPriceWithVatDTO.cpp


#include "GetPriceWithVatDTO.h"

using namespace Tiny;

GetPriceWithVatDTO::GetPriceWithVatDTO()
{
	value = float(0);
	discountBase = float(0);
	currencyId = CurrencyType();
	vat = int(0);
	updatedAt = std::string();
}

GetPriceWithVatDTO::GetPriceWithVatDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

GetPriceWithVatDTO::~GetPriceWithVatDTO()
{

}

void
GetPriceWithVatDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *valueKey = "value";

    if(object.has_key(valueKey))
    {
        bourne::json value = object[valueKey];



        jsonToValue(&value, value, "long");


    }

    const char *discountBaseKey = "discountBase";

    if(object.has_key(discountBaseKey))
    {
        bourne::json value = object[discountBaseKey];



        jsonToValue(&discountBase, value, "long");


    }

    const char *currencyIdKey = "currencyId";

    if(object.has_key(currencyIdKey))
    {
        bourne::json value = object[currencyIdKey];




        CurrencyType* obj = &currencyId;
		obj->fromJson(value.dump());

    }

    const char *vatKey = "vat";

    if(object.has_key(vatKey))
    {
        bourne::json value = object[vatKey];



        jsonToValue(&vat, value, "int");


    }

    const char *updatedAtKey = "updatedAt";

    if(object.has_key(updatedAtKey))
    {
        bourne::json value = object[updatedAtKey];



        jsonToValue(&updatedAt, value, "std::string");


    }


}

bourne::json
GetPriceWithVatDTO::toJson()
{
    bourne::json object = bourne::json::object();





    object["value"] = getValue();






    object["discountBase"] = getDiscountBase();







	object["currencyId"] = getCurrencyId().toJson();





    object["vat"] = getVat();






    object["updatedAt"] = getUpdatedAt();



    return object;

}

long
GetPriceWithVatDTO::getValue()
{
	return value;
}

void
GetPriceWithVatDTO::setValue(long  value)
{
	this->value = value;
}

long
GetPriceWithVatDTO::getDiscountBase()
{
	return discountBase;
}

void
GetPriceWithVatDTO::setDiscountBase(long  discountBase)
{
	this->discountBase = discountBase;
}

CurrencyType
GetPriceWithVatDTO::getCurrencyId()
{
	return currencyId;
}

void
GetPriceWithVatDTO::setCurrencyId(CurrencyType  currencyId)
{
	this->currencyId = currencyId;
}

int
GetPriceWithVatDTO::getVat()
{
	return vat;
}

void
GetPriceWithVatDTO::setVat(int  vat)
{
	this->vat = vat;
}

std::string
GetPriceWithVatDTO::getUpdatedAt()
{
	return updatedAt;
}

void
GetPriceWithVatDTO::setUpdatedAt(std::string  updatedAt)
{
	this->updatedAt = updatedAt;
}



